SELF_HOSTED_SETUP.md

```diff
@@ -0,0 +1,127 @@
+# Self-hosted Kubernetes setup
+
+If you are running your own Kubernetes cluster, there are several steps required for this feature to work.
+
+This feature requires Kubernetes 1.12 or greater.
+
+## Projected Token Signing Keypair
+
+The first thing required is a new key pair for signing and verifying projected
+service account tokens. This can be done using the following `ssh-keygen`
+commands.
+
+```bash
+# Generate the keypair
+PRIV_KEY="sa-signer.key"
+PUB_KEY="sa-signer.key.pub"
+PKCS_KEY="sa-signer-pkcs8.pub"
+# Generate a key pair
+ssh-keygen -t rsa -b 2048 -f $PRIV_KEY -m pem
+# convert the SSH pubkey to PKCS8
+ssh-keygen -e -m PKCS8 -f $PUB_KEY > $PKCS_KEY
+```
+
+## Public Issuer
+
+As of 1.16, Kubernetes does not include an OIDC discovery endpoint itself (see
+[kubernetes/community#1190](https://github.com/kubernetes/enhancements/pull/1190)),
+so you will need to put your public signing key somewhere that AWS STS can
+discover it. This example, we will create one in a public S3 bucket, but you
+could host the following documents any way you'd like on a different domain.
```

Comment:
any where?

Reply:
Eh, either or. I was probably thinking “any way” like dynamically instead of a static JSON document